### ISSUE TITLE
cpi property stemcell_os_type should be configurable to leverage PVHVM feature for linux kernel 2.6.32+

### ISSUE THREAD

**Issue by f-guichard** [cloudfoundry-community/bosh-cloudstack-cpi-release#29]
Enabling PVHVM guests in XenServer 6.5 requires that guest templates are registered under specific template name "Ubuntu 14.04". Still, for XenServer 6.2, enabling PV Drivers still needs "Other PV (64 Bits)" template name.

So it might be a good idea to externalise this attribute as a stemcell property or include a specific logic to compute this attribute value in cpi itself.

**Comment by poblin-orange** [comment]
see
 http://support.citrix.com/article/CTX121875
http://wiki.xen.org/wiki/Xen_Linux_PV_on_HVM_drivers
for reference. 
this is to be used with  #25

**Comment by poblin-orange** [comment]
Adding stemcell_os_type property for template publication. With the stemcell_use_hvm boolean to true, this should trigger cloudstack pvhm on the vms.
NB1: default is still Other PV 64 bits, for older xen compatibility
NB2: setting this property for a give Os type (eg Ubuntu 14.04) will lock the bosh director on a type of stemcell (cant pusblish a centos stemcell for instance), as it is a cpi level configuration.

**Comment by poblin-orange** [comment]
error when attaching disk with template registered with hvm flag

cpi-core log 

```
Failed to attach volume for uuid: 188907a9-3e43-4b59-8cd5-514f0028b788 due to You attempted an operation on a VM which requires PV drivers to be installed but the drivers were not detected.
You attempted an operation on a VM which requires PV drivers to be installed but the drivers were not detected.
```

Source tracking: 
 https://github.com/apache/cloudstack/blob/4.7/plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/XenServerStorageProcessor.java

```
     <dependency>
      <groupId>net.java.dev.vcc.thirdparty</groupId>
      <artifactId>xen-api</artifactId>
      <version>${cs.xapi.version}</version>
    </dependency>
```

 https://maven-repository.com/artifact/net.java.dev.vcc.thirdparty/xen-api/6.2.0-3.1

```
     /**
     * Hotplug the specified VBD, dynamically attaching it to the running VM
     * First published in XenServer 4.0.
     *
     */
    public void plug(Connection c) throws
       BadServerResponse,
       XenAPIException,
       XmlRpcException {
        String method_call = "VBD.plug";
        String session = c.getSessionReference();
        Object[] method_params = {Marshalling.toXMLRPC(session), Marshalling.toXMLRPC(this.ref)};
        Map response = c.dispatch(method_call, method_params);
        return;
    }
```

 http://xenserver.org/overview-xenserver-open-source-virtualization/download.html

 http://xenserver.org/partners/developing-products-for-xenserver/20-dev-hints/90-xs-log-debug-understand.html
 http://docs.citrix.com/content/dam/en-us/xenserver/xenserver-65/xenserver65sp1_management_api_guide.pdf

```
 RPC name: plug
Overview:
Hotplug the specified VBD, dynamically attaching it to the ru
nning VM.
Signature:
void plug (session_id s, VBD ref self)
Arguments:
type
name
description
VBD ref
self
The VBD to hotplug
Return Type:
void
```

http://xenproject.org/developers/teams/xapi.html

https://github.com/xapi-project/xen-api/tree/master/ocaml/xapi

https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/xapi_vbd.ml

https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/helpers.ml  

```
(** Returns true if the supplied VM configuration is HVM.
    NB that just because a VM's current configuration looks like HVM doesn't imply it
    actually booted that way; you must check the boot_record to be sure *)
let is_hvm (x: API.vM_t) = x.API.vM_HVM_boot_policy <> ""

let will_boot_hvm ~__context ~self = Db.VM.get_HVM_boot_policy ~__context ~self <> ""

let has_booted_hvm ~__context ~self =
  let boot_record = get_boot_record ~__context ~self in
  boot_record.API.vM_HVM_boot_policy <> ""
```

**Comment by poblin-orange** [comment]
manually mounting the ephemeral disk from cloudstack portal is ok after failure.
It seems there is a delay between instance boot and xenserver pvhvm detection.
Adding a new property to add some delay after vm creation helps =>  test ok on ikoula with a value of 15s (default cpi bosh release value  new set to 10s)
